homeassistant/components/watergate/entity.py

```diff
@@ -0,0 +1,52 @@
+"""Watergate Base Entity Definition."""
+
+from dataclasses import dataclass
+
+from watergate_local_api import WatergateLocalApiClient
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_NAME
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import DOMAIN, MANUFACTURER
+from .coordinator import WatergateDataCoordinator
+
+type WatergateConfigEntry = ConfigEntry[WatergateData]
+
+
+class WatergateEntity(CoordinatorEntity[WatergateDataCoordinator]):
+    """Define a base Watergate entity."""
+
+    def __init__(
+        self,
+        coordinator: WatergateDataCoordinator,
+        entry: WatergateConfigEntry,
+        entity_name: str,
+    ) -> None:
+        """Initialize the entity."""
+        super().__init__(coordinator)
+        self._attr_has_entity_name = True
+        self._api_client = coordinator.api
+        self._attr_unique_id = f"{entry.data[CONF_NAME]}.{entity_name}"
```

Comment:
will be unique across platforms anyways, no need to add the entity name here

Reply:
I am not sure about that. What if someone will have two devices with the same name? This CONF_NAME is something that the user is setting up.

Reply:
Ok. It works. But I am not sure what will happen if we will add more entities. I will have to figure it later.

Reply:
Yes, I got it. This is what I did, but still, I am concerned about what will happen in the future. After this PR, I have more ;) Currently, it works perfectly so there is no need to think about the near future. I will refactor it later.

Reply:
CONF_NAME is not good here. If you have no other ways to generate a unique id here (e.g. serial number of the device), `entry.entry_id` is the fallback option to make sure it's unique

Reply:
Also refactoring this is not really possible since this you would mean a breaking change, so let's get this right now

Reply:
Ok, let me know if you think something else needs to be changed. I do not have anything specific for this device. So, I guess entry_id should be here.

Reply:
But, should I use the entity_name or just entry_id? Entry_id will not be unique across all my later entities, right?

Reply:
Yes, I was passing only valve constant since this is the only entity I have. I cannot have more than one platform. But I will have 20 other entities for that device, so I guess I should revert that entity_name passing, right?

Reply:
Yeah apologies

Reply:
Done. Hopefully, it is everything.